packages/flutter_tools/test/general.shard/project_file_invalidator_test.dart

```diff
@@ -3,67 +3,78 @@
 // found in the LICENSE file.
 
 import 'package:file/memory.dart';
-import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/base/platform.dart';
 import 'package:flutter_tools/src/run_hot.dart';
-import 'package:meta/meta.dart';
 
 import '../src/common.dart';
-import '../src/context.dart';
 
 // assumption: tests have a timeout less than 100 days
 final DateTime inFuture = DateTime.now().add(const Duration(days: 100));
+final DateTime inPast = DateTime.now().subtract(const Duration(days: 100));
 
 void main() {
-  group('ProjectFileInvalidator', () {
-    _testProjectFileInvalidator(asyncScanning: false);
-  });
-  group('ProjectFileInvalidator (async scanning)', () {
-    _testProjectFileInvalidator(asyncScanning: true);
+  FakePlatform platform;
+  BufferLogger logger;
+
+  setUp(() {
+    platform = FakePlatform();
+    logger = BufferLogger();
   });
-}
 
-void _testProjectFileInvalidator({@required bool asyncScanning}) {
-  const ProjectFileInvalidator projectFileInvalidator = ProjectFileInvalidator();
+  for (bool asyncScanning in <bool>[true, false]) {
+    test('No last compile', () async {
```

Comment:
should these two tests have unique names (i.e. interpolate `asyncScanning`)?

Reply:
Good call, fixed